ProjectLighthouse.Servers.Website/Pages/SlotPage.cshtml.cs

```diff
@@ -37,22 +39,28 @@ public async Task<IActionResult> OnGet([FromRoute] int id)
         // Determine if user can view slot according to creator's privacy settings
         if (this.User == null || !this.User.IsAdmin)
         {
-            switch (slot.Creator.ProfileVisibility)
+            switch (slot.Creator.LevelVisibility)
```

Comment:
This entire calculation of whether a profile should be private or not could be refactored into its own function that takes a privacy level and a user and outputs a bool